### PR TITLE
Initial group agreement management UI

### PR DESCRIPTION
**Pull request by nicksellen**
Implements one part of group agreements - showing an alert when the user has not agreed and letting them agree :)

You can try it out at https://group-agreements--karrot.netlify.com/#/group/1/wall (only group 1 has an agreement set).

You cannot edit the agrerements in the UI yet. In progress...

(note: TERMS OF SERVICE comes from the user specified values in the agreement, it would be whatever the group agreement manager set it to...)

agreements

### CODE REVIEW

**Comment by nicksellen** [comment]
... and with the dialog opened...

agreements-open

**Comment by nicksellen** [comment]
Ok, agreement management added!
- create initial agreement
- minor edit to agreement (users don't need to re-agree)
- major change to agreement (users need to re-agree)
- remove agreement

Currently nothing is actually restricted on the frontend or the backend, so that is next.

Although this is useful/mergable as it is, so would propose to continue work in future PRs.

**Comment by nicksellen** [comment]
Oh, todo:
- [ ] hide "Minor edit?" checkbox when creating initial agreement

**Comment by nicksellen** [comment]
@stefan-fstaiwan / @Sliverriver /  @brnsolikyl would appreciate your feedback too! Would be nice to know if this will help progress your foodsaving movements :)

You can try it out at https://group-agreements--karrot.netlify.com/ using the same credentials as you might have for any of the dev sites (https://dev.foodsaving.world/ / https://alpha.foodsaving.world/ / https://app.foodsaving.world/), and create a new account if you don't have one, or use foo@foo.com / foofoo.

Although, what is not implemented yet is: _ 3.2 If users join a group, joining and accepting should be done in one step_ (from https://github.com/yunity/karrot-frontend/issues/324#issuecomment-312685256).

**Comment by nicksellen** [comment]
... although I just realised that as there is no UI for role management yet, you can't manage the agreement anyway :) ... but if you want to try it out, let me know, and I can manually give you the correct role...

File: src/components/Group/GroupEdit.vue [hunk 1]
```diff
@@ -183,6 +183,10 @@ export default {
         this.$emit('save', { ...objectDiff(this.group, this.groupEdit), id: this.group.id }, event)
       }
     },
+    timezoneFilter (terms, { field, list }) {
```
Comment (tiltec) [inline src/components/Group/GroupEdit.vue]: seems unrelated to the PR, does it enable more relaxed filtering of timezones? I think it's the same behavior as in the angular code, so it's good.

File: src/store/modules/agreements.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    commit(types.REQUEST_ENTRY)
+    commit(types.RECEIVE_ENTRY, { agreement: await agreements.get(id) })
+  },
+  async agree ({ commit }, id) {
+    commit(types.REQUEST_AGREE)
+    commit(types.RECEIVE_ENTRY, { agreement: await agreements.agree(id) })
+  },
+  async create ({ commit }, agreement) {
+    commit(types.REQUEST_CREATE)
+    agreement = await agreements.create(agreement)
```
Comment (tiltec) [inline src/store/modules/agreements.js]: should handle server validation errors, maybe track the to do tasks in an issue?
by the way, I would rename the `onlyHandleAPIError` helper to `handleServerValidationError`, what do you think about it?

File: src/components/Group/GroupEdit.vue (hunk 1 above)
Comment (nicksellen) [reply src/components/Group/GroupEdit.vue]: Yes, I tried typing "berlin" and it didn't find `Europe/Berlin`. And yes it's unrelated, just seemed too small for the extra PR bureaucracy.

File: src/components/Group/AgreementForm.vue [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+          :autofocus="true"
+          @blur="$v.agreementEdit.title.$touch"
+          autocomplete="off"
+        />
+      </q-field>
+
+      <q-field
+        icon="fa-file-text-o"
+        :label="$t('AGREEMENT.CONTENT')"
+        :helper="$t('AGREEMENT.CONTENT_HELPER')">
```
[the remark this answers is not in the document]
Comment (nicksellen) [reply src/components/Group/AgreementForm.vue]: I think we are undoing each others changes then :)

I notice quasar does the on-a-new-line style, personally I don't like it though. I wonder if there vuetemplatelint we can use...

File: src/components/Group/AgreementForm.vue [hunk 4]
```diff
@@ -0,0 +1,158 @@
+<template>
+  <div class="edit" :class="{ changed: hasChanged }">
+    <form @submit="save">
+      <q-field
+        icon="fa-star"
+        :label="$t('AGREEMENT.TITLE')"
+        :helper="$t('AGREEMENT.TITLE_HELPER')">
```
[the remark this answers is not in the document]
Comment (nicksellen) [reply src/components/Group/AgreementForm.vue]: No, I'm very naughty...

**Comment by stefan-fstaiwan** [comment]
I would like to try it out, but I'm confused now, how to do it...

**Comment by nicksellen** [comment]
@stefan-fstaiwan I wrote to you in slack, can chat you through it (I need to do a manual step for you...)
